### PR TITLE
Migrations

### PR DESCRIPTION
**Pull request by macfarla**
make migrations work with already deployed ingress

### CODE REVIEW

File: migrations/2_deploy_ingress_contract.js [hunk 1]
```diff
@@ -12,11 +12,12 @@ module.exports = (deployer, network) => {
 
     } else {
         // If supplied an address, make sure there's something there
-        const ingressInstance = Ingress.at(ingressAddress);
+        let ingressInstance = await Ingress.at(ingressAddress);
```
Comment (Errorific) [inline migrations/2_deploy_ingress_contract.js]: These can be const, do we want them to be const?

Comment (macfarla) [reply migrations/2_deploy_ingress_contract.js]: I missed a commit, in which I make it not const - to accommodate migrating to ganache where it is not fixed

Comment (macfarla) [reply migrations/2_deploy_ingress_contract.js]: const it is

File: migrations/2_deploy_ingress_contract.js [hunk 2]
```diff
@@ -1,23 +1,35 @@
...
-    } else {
+    } 
         // If supplied an address, make sure there's something there
-        const ingressInstance = Ingress.at(ingressAddress);
+        const ingressInstance = await Ingress.at(ingressAddress);
         try {
-            ingressInstance.getContractVersion();
-            console.log("   > Ingress contract previously initialised at address = " + ingressAddress);
-        } catch {
+            let result = await ingressInstance.getContractVersion();
```
Comment (Errorific) [inline migrations/2_deploy_ingress_contract.js]: const here too

File: migrations/2_deploy_ingress_contract.js [hunk 3]
```diff
@@ -1,23 +1,35 @@
...
+const rulesContractName = Web3Utils.utf8ToHex("rules");
 
 /* The address of the ingress contract if pre deployed */
 var ingressAddress = process.env.INGRESS_CONTRACT_ADDRESS;
 
-module.exports = (deployer, network) => {
+module.exports = async(deployer, network) => {
     if (! ingressAddress) {
         // Only deploy if we haven't been provided a predeployed address
         deployer.deploy(Ingress).then(() => {
```
Comment (Errorific) [inline migrations/2_deploy_ingress_contract.js]: I think the error you're getting out of CI might be due to this still being a promise style call when the function is marked async. The test is racing ahead before the Ingress has finished deploying

File: migrations/2_deploy_ingress_contract.js [hunk 4]
```diff
@@ -1,23 +1,35 @@
...
             console.log("   > Deployed Ingress contract to address = " + Ingress.address);
         });
+        ingressAddress = Ingress.address;
 
-    } else {
+    } 
         // If supplied an address, make sure there's something there
-        const ingressInstance = Ingress.at(ingressAddress);
+        const ingressInstance = await Ingress.at(ingressAddress);
```
Comment (Errorific) [inline migrations/2_deploy_ingress_contract.js]: Indent is skewed after removing the if/else

**Pull request merged**